pkg/operator/controller/certificate-publisher/publish_certs.go

```diff
@@ -68,7 +68,19 @@ func desiredRouterCertsGlobalSecret(secrets []corev1.Secret, ingresses []operato
 
 	ingressToSecret := map[*operatorv1.IngressController]*corev1.Secret{}
 	for i, ingress := range ingresses {
-		name := controller.RouterEffectiveDefaultCertificateSecretName(&ingress, operandNamespace)
+		// Check if ingress.Spec.DefaultCertificate is an available secret
+		// in the secrets slice. If it is not, attempt to fall back to the
+		// operator generated default certificate. If ingress.Spec.DefaultCertificate
+		// is updated to point to a non-existent secret, the certificate controller
+		// will not delete the operator generated default certificate.
+		// See https://bugzilla.redhat.com/show_bug.cgi?id=1887441
```

Comment:
It would be a good idea to log the issue with non-existent secret, because if there was a legitimate reason to change it and it couldn't be changed due to a typo for example, the user would want to know that.

Reply:
The issue gets logged by the default certificate controller's reconcile loop here: https://github.com/openshift/cluster-ingress-operator/pull/505/files#diff-16d9c01d5eac10bd8a3012d74ef5feb542287746e676beca44479eb80cbaa124R36

Does that seem sufficient in this case?